org.jacoco.core.test.validation.groovy/pom.xml

```diff
@@ -0,0 +1,89 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+   Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+   All rights reserved. This program and the accompanying materials
+   are made available under the terms of the Eclipse Public License v1.0
+   which accompanies this distribution, and is available at
+   http://www.eclipse.org/legal/epl-v10.html
+
+   Contributors:
+      Andres Almiray - initial API and implementation
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.jacoco</groupId>
+        <artifactId>org.jacoco.core.test.validation</artifactId>
+        <version>0.8.3-SNAPSHOT</version>
+        <relativePath>../org.jacoco.core.test.validation</relativePath>
+    </parent>
+
+    <artifactId>org.jacoco.core.test.validation.groovy</artifactId>
+
+    <name>JaCoCo :: Test :: Core :: Validation Groovy 2</name>
+
+    <properties>
+        <gmaven.version>1.6</gmaven.version>
+        <groovy.version>2.5.3</groovy.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>org.jacoco.core.test</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.codehaus.groovy</groupId>
+            <artifactId>groovy-all</artifactId>
```

Comment:
`groovy-all` brings quite a lot of dependencies - `mvn -pl org.jacoco.core.test.validation.groovy dependency:tree`:

```
[INFO] \- org.codehaus.groovy:groovy-all:pom:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-ant:jar:2.5.3:compile
[INFO]    |  +- org.apache.ant:ant:jar:1.7.1:compile (version managed from 1.9.13)
[INFO]    |  |  \- org.apache.ant:ant-launcher:jar:1.9.13:compile
[INFO]    |  +- org.apache.ant:ant-junit:jar:1.7.1:runtime (version managed from 1.9.13)
[INFO]    |  \- org.apache.ant:ant-antlr:jar:1.9.13:runtime
[INFO]    +- org.codehaus.groovy:groovy-cli-commons:jar:2.5.3:compile
[INFO]    |  \- commons-cli:commons-cli:jar:1.4:compile
[INFO]    +- org.codehaus.groovy:groovy-cli-picocli:jar:2.5.3:compile
[INFO]    |  \- info.picocli:picocli:jar:3.6.0:compile
[INFO]    +- org.codehaus.groovy:groovy-console:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-datetime:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-docgenerator:jar:2.5.3:compile
[INFO]    |  \- com.thoughtworks.qdox:qdox:jar:1.12.1:compile
[INFO]    +- org.codehaus.groovy:groovy-groovydoc:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-groovysh:jar:2.5.3:compile
[INFO]    |  \- jline:jline:jar:2.14.6:compile
[INFO]    +- org.codehaus.groovy:groovy-jmx:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-json:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-jsr223:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-macro:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-nio:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-servlet:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-sql:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-swing:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-templates:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-test:jar:2.5.3:compile
[INFO]    +- org.codehaus.groovy:groovy-test-junit5:jar:2.5.3:compile
[INFO]    |  +- org.junit.platform:junit-platform-launcher:jar:1.3.1:compile
[INFO]    |  |  +- org.apiguardian:apiguardian-api:jar:1.0.0:compile
[INFO]    |  |  \- org.junit.platform:junit-platform-engine:jar:1.3.1:compile
[INFO]    |  |     +- org.junit.platform:junit-platform-commons:jar:1.3.1:compile
[INFO]    |  |     \- org.opentest4j:opentest4j:jar:1.1.1:compile
[INFO]    |  \- org.junit.jupiter:junit-jupiter-engine:jar:5.3.1:runtime
[INFO]    |     \- org.junit.jupiter:junit-jupiter-api:jar:5.3.1:runtime
[INFO]    +- org.codehaus.groovy:groovy-testng:jar:2.5.3:compile
[INFO]    |  \- org.testng:testng:jar:6.13.1:runtime
[INFO]    |     \- com.beust:jcommander:jar:1.72:runtime
[INFO]    \- org.codehaus.groovy:groovy-xml:jar:2.5.3:compile
```

`maven-surefire-plugin` sees `org.testng:testng` and uses it for execution - stacktrace of failed test:
```
at org.testng.TestNG.run(TestNG.java:996)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:112) 
```

so most likely `maven-surefire-plugin` version `2.9` declared in our parent pom doesn't work because of https://issues.apache.org/jira/browse/SUREFIRE-1019

I think that in any case just `groovy` should be used instead of `groovy-all`.

Reply:
OK